tests/middleware/test_middleware.py

```diff
@@ -12,11 +12,25 @@ async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
         await self.app(scope, receive, send)
 
 
-def test_middleware_repr() -> None:
-    middleware = Middleware(CustomMiddleware, "foo", bar=123)
-    assert repr(middleware) == "Middleware(CustomMiddleware, 'foo', bar=123)"
+def test_class_matches_middleware_protocol() -> None:
+    assert isinstance(CustomMiddleware, _MiddlewareFactory)
```

Comment:
Why are you deleting the old test and adding a new seemingly unrelated test? I don't think this test is valuable. If you wanted to check this do it via type checking: `foo: _MiddlewareFactory = CustomMiddleware`. But I don't think that is very valuable: `CustomMiddleware` is a test only class, we don't need to test it directly only indirectly via it's involvement in other tests